RecoTauTag/RecoTau/plugins/PATTauIDEmbedder.cc

```diff
@@ -0,0 +1,95 @@
+#include "FWCore/Framework/interface/stream/EDProducer.h"
+#include "FWCore/Framework/interface/Event.h"
+#include "FWCore/Framework/interface/EventSetup.h"
+#include "FWCore/Utilities/interface/InputTag.h"
+#include "FWCore/ParameterSet/interface/ParameterSet.h"
+#include "DataFormats/PatCandidates/interface/PATTauDiscriminator.h"
+#include "DataFormats/PatCandidates/interface/Tau.h"
+#include "FWCore/Utilities/interface/transform.h"
+
+class PATTauIDEmbedder : public edm::stream::EDProducer<>
+{
+public:
+
+	explicit PATTauIDEmbedder(const edm::ParameterSet&);
+	~PATTauIDEmbedder(){};
+
+	void produce(edm::Event&, const edm::EventSetup&);
+
+private:
+
+//--- configuration parameters
+	edm::EDGetTokenT<pat::TauCollection> src_;
+	typedef std::pair<std::string, edm::InputTag> NameTag;
+	std::vector<NameTag> tauIDSrcs_;
+	std::vector<edm::EDGetTokenT<pat::PATTauDiscriminator> > patTauIDTokens_;
+};
+
+PATTauIDEmbedder::PATTauIDEmbedder(const edm::ParameterSet& cfg)
+{
+	src_ = consumes<pat::TauCollection>(cfg.getParameter<edm::InputTag>("src"));
+	// it might be a single tau ID
+	if (cfg.existsAs<edm::InputTag>("tauIDSource")) {
+		tauIDSrcs_.push_back(NameTag("", cfg.getParameter<edm::InputTag>("tauIDSource")));
+	}
+	// or there might be many of them
+	if (cfg.existsAs<edm::ParameterSet>("tauIDSources")) {
```

Comment:
I'm not sure what you are referring to but I'll try to clarify further: 

(1)
if you compare the lines above to https://github.com/cms-sw/cmssw/blob/master/PhysicsTools/PatAlgos/plugins/PATTauProducer.cc#L74 ff you will see that the access is exactly the same. 

(2)
You can see from https://github.com/cms-sw/cmssw/blob/master/PhysicsTools/PatAlgos/plugins/PATTauProducer.cc#L319 how the discriminator values are linked to the name tags as given in  the cfi. 

(3)
You can find how the specific name tag is called here https://github.com/cms-sw/cmssw/blob/master/DataFormats/PatCandidates/interface/Tau.h#L369.